includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -1460,20 +1439,112 @@ protected function build_shipping_methods( $shipping_methods ) {
 		return $shipping;
 	}
 
+
+	/**
+	 * Builds the response object to pass to Payment Request
+	 *
+	 * @since 5.3.0
+	 *
+	 * @param bool   $shipping_pending True is the user hasn't defined yet a shipping address.
+	 * @param string $page             Page where the request was made. It can be `product_page` or `cart_page`.
+	 *
+	 * @return array Data object used to update Payment Request button on the client.
+	 */
+	protected function build_response( $shipping_pending = true, $page = 'product_page' ) {
+		if ( 'product_page' == $page ) {
+			$itemized_display_items = true;
+		} elseif ( 'cart_page' == $page ) {
+			$itemized_display_items = false;
+		}
+
+		$data                 = [];
+		$data['currency']     = strtolower( get_woocommerce_currency() );
+		$data['country_code'] = substr( get_option( 'woocommerce_default_country' ), 0, 2 );
+
+		// Taxes
+		$tax              = WC()->cart->tax_total + WC()->cart->shipping_tax_total;
+		$tax_to_substract = 0;
+
+		if ( wc_tax_enabled() ) {
+			if ( $shipping_pending ) {
+				$tax_to_substract = $tax;
+				$tax              = 0;
+			}
+		}
+
+		// Shipping
+		$shipping              = wc_format_decimal( WC()->cart->get_shipping_total(), WC()->cart->dp );
+		$shipping_to_substract = 0;
+
+		if ( wc_shipping_enabled() && WC()->cart->needs_shipping() ) {
+			$data['requestShipping'] = ( wc_shipping_enabled() && WC()->cart->needs_shipping() );
+
+			if ( $shipping_pending ) {
+				$shipping_to_substract = $shipping;
+				$shipping              = 0;
+
+				$data['shippingOptions'] = [
+					'id'     => 'pending',
+					'label'  => __( 'Pending', 'woocommerce-gateway-stripe' ),
+					'detail' => '',
+					'amount' => 0,
+				];
+			}
```

Comment:
Is this `$shipping_pending` parameter really necessary? Why do we need it?

Reply:
I'm asking because the `pending` prop for the labels was added statically before, and that made the implementation much simpler. I wonder what makes it different having it based on `$shipping_pending` now.

Reply:
> Is this $shipping_pending parameter really necessary? Why do we need it?

`$shipping_pending` indicates if the user has selected or entered a shipping address in the payment dialog. If the user hasn't selected anything (`$shipping_pending == true`) we should return the shipping item as empty.

>I'm asking because the pending prop for the labels was added statically before, and that made the implementation much simpler. I wonder what makes it different having it based on $shipping_pending now.

Could you elaborate on this. How it was simpler? What do you mean with `pending prop for the labels`?

> I wonder what makes it different having it based on $shipping_pending now.

I think it makes sense to mark the shipping amount as `pending=true` if the user has not picked or entered an address on the payment dialog. IMO it's not pending anymore after the user has provided an address but I guess the amount is never settled since it could change based on the address or the shipping method. Let me know if you want me to change it to always return `pending=true` if that's what you meant.

On the other hand it looks like the `pending` attribute doesn't have real usage https://stripe.com/docs/js/appendix/payment_item_object#payment_item_object-pending

Reply:
> On the other hand it looks like the pending attribute doesn't have real usage

Exactly, I'm wondering why are we passing this property as a dynamic value now if it doesn't even change the Payment Request API flow. Or it does?

> Could you elaborate on this. How it was simpler? What do you mean with pending prop for the labels?

I mean, in the current implementation we're [passing the pending prop statically](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/1fc58853fa3c5c64cf1635faf443ceeaa368ff0e/includes/payment-methods/class-wc-stripe-payment-request.php#L364) to the `Tax` and `Shipping` labels, which is much simpler because we don't need the `$shipping_pending` variable and we don't need to pass it from the front-end either.

I noticed you added this variable, which we didn't have before, so I'm asking because I didn't understand what's the real purpose of it.